### PR TITLE
added signup login logout

### PR DESCRIPTION
**Pull request by Hnh013**
pages added : Signup , Login
added routing and links for both pages

### CODE REVIEW

**Review by hrishi82 — APPROVED** [review]
code seems good to me, can be merged

File: src/frontend/styles/pages/auth.css [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+    justify-content: center;
+    text-align: center;
+}
+
+.err-msg {
+    font-size: 0.875rem;
+  display: none;
+}
+
+input:invalid ~ span.err-msg {
```
Comment (SJTGSHIVAM) [inline src/frontend/styles/pages/auth.css:58]: Good usage of css

File: src/frontend/templates/components/Navbar.js [hunk 2]
```diff
@@ -1,8 +1,19 @@
 import React, { useState } from 'react';
 import '../../styles/components/navbar.css';
```
Comment (SJTGSHIVAM) [inline src/frontend/templates/components/Navbar.js:2]: Do check how to configure absolute routing it makes like a lot easier.

File: src/frontend/templates/components/Navbar.js [hunk 3]
```diff
@@ -1,8 +1,19 @@
...
-import { Link } from 'react-router-dom';
+import { Link , useNavigate } from 'react-router-dom';
+import { useUser } from '../../contexts/userContext';
+import { toastSuccess } from '../../services/toastService';
 
 export const Navbar = () => {
+  const { userState , userDispatcher } = useUser();
+  const navigate = useNavigate();
+
+  const logoutUser = async (e) => {
```
Comment (SJTGSHIVAM) [inline src/frontend/templates/components/Navbar.js]: e is passed but not ussed. this creates a little confusion.

File: src/frontend/templates/pages/Login.js [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+            navigate('/');
+        } else {
+            if(response.actionResponse.response.status === 404 || response.actionResponse.response.status === 401 ) {
+            toastError(response.actionResponse.response.data.errors[0].split('.')[0]); 
+           } else {
+            toastError('Oops, an error has occured!');
+           }
+        }
+    } catch(error) {
+        response = error;
```
Comment (SJTGSHIVAM) [inline src/frontend/templates/pages/Login.js:39]: please avoid doing this naming it creates a little confusion

**Review by SJTGSHIVAM — APPROVED** [review]
Code looks good. I don't see any major issues. I have left some comments Please check those. We are good to go forward with this.

File: src/frontend/templates/components/Navbar.js (hunk 3 above)
Comment (Hnh013) [reply src/frontend/templates/components/Navbar.js]: hi shivam ,sure will remove e in cases its not used

File: src/frontend/templates/pages/Login.js (hunk 4 above)
Comment (Hnh013) [reply src/frontend/templates/pages/Login.js:39]: thanks for this shivam , will try to better my naming convention

**Pull request merged**